### PR TITLE
Update to apollo codegen 0.15.0

### PR DESCRIPTION
**Pull request by BenSchwab**
This is a pre-req for https://github.com/apollographql/apollo-android/issues/550

This includes a change https://github.com/apollographql/apollo-codegen/commit/faa5715aee550f5f1adb04ce08ed0ed523accf8e which enables:

`mergeInFieldsFromFragmentSpreads`.

### CODE REVIEW

File: apollo-integration/src/test/java/com/apollographql/apollo/ResponseWriteTestCase.java [hunk 1]
```diff
@@ -249,35 +250,63 @@ public void operation_with_fragments() throws Exception {
     assertThat(hero.fragments().heroWithFriendsFragment().friends().get(2).fragments().humanWithIdFragment().id()).isEqualTo("1003");
     assertThat(hero.fragments().heroWithFriendsFragment().friends().get(2).fragments().humanWithIdFragment().name()).isEqualTo("Leia Organa");
 
+    final HumanWithIdFragment supermanAsHumanFragment = new HumanWithIdFragment(
```
Comment (BenSchwab) [inline apollo-integration/src/test/java/com/apollographql/apollo/ResponseWriteTestCase.java]: @sav007 This is an example of https://github.com/apollographql/apollo-android/issues/559

The fact that fragment types are merged in now (and a different type) makes the imperative store api a bit more clumsy and the potential problem -- what happens if a user inconsistently creates the two different fragment objects? Which one will win in the merge process?

Comment (sav007) [inline apollo-integration/src/test/java/com/apollographql/apollo/ResponseWriteTestCase.java]: Yeah, I have the same thought while I was implementing write operation. I think it is user responsibility to make sure that fragments with the same id have the same data. If they are different then it depends of field tree traversal

**Comment by sav007** [comment]
I see that fields from fragment being merged into query, is it because of logic behind operation id generation?

File: apollo-compiler/src/test/graphql/com/example/fragment_in_fragment/TestOperation.graphql [hunk 2]
```diff
@@ -14,7 +14,7 @@ fragment starshipFragment on Starship {
   pilotConnection {
     edges {
       node {
-        ...pilotFragment
+       name
```
Comment (sav007) [inline apollo-compiler/src/test/graphql/com/example/fragment_in_fragment/TestOperation.graphql]: Is it intentionally? As this test was used for testing fragment inside another fragment

Comment (BenSchwab) [inline apollo-compiler/src/test/graphql/com/example/fragment_in_fragment/TestOperation.graphql]: Good catch, was testing something myself and forgot to switch back

**Comment by BenSchwab** [comment]
@sav007 The field merging is separate from operation id generation. @martijnwalraven May be able to give more insight on it.

I will follow up with another PR to enable the operationId flag, but I wanted to do this PR separately to discuss the field spreading, in case there is any changes we want to do to handle it.

**Comment by BenSchwab** [comment]
https://github.com/apollographql/apollo-codegen/pull/147 This is the PR that enables operationId in Apollo codegen. When we turn the flag on we will get an operationId and source with fragments merged on the IR file (this is different than the fragment spreading we see in this PR) and a build time generated json map file of operationId -> source with fragments

**Comment by BenSchwab** [comment]
In 0.15.2 we can turn off the fragment spread. Looks like there will be no IR file changes until I enable the operationId generation. So going to close this, and proceed work on the operationId change.

**Pull request closed without merging**